internal/config/config.go

```diff
@@ -1,8 +1,29 @@
+// Copyright 2020 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
 package config
 
 import (
+	"path"
+
 	"github.com/layer5io/meshery-adapter-library/config"
 	"github.com/layer5io/meshery-adapter-library/config/provider"
+	"github.com/layer5io/meshkit/utils"
+)
+
+var (
+	ConfigRootPath = path.Join(utils.GetHome(), ".meshery")
```

Comment:
For my personal, we need add the code comment on this. In fact, I hit the issue on this, if I did not create the folder in the main function, there will be threw an error with viper.

Reply:
```
func init() {
	err := os.MkdirAll(path.Join(utils.GetHome(), ".meshery", "bin"), 0750)
	if err != nil {
		fmt.Println(err)
		os.Exit(0)
	}
}
```

Reply:
> For my personal, we need add the code comment on this. In fact, I hit the issue on this, if I did not create the folder in the main function, there will be threw an error with viper.

I hit the same issue when the Viper implementation in mesh-adapter-library was changed.
I agree, this should be documented, preferably in the meshery-adapter-library - assuming that this is the first place developers will look for documentation. But of course it could in addition also be mentioned here.
I feel there should be more documentation in general in the common libraries. It would also be helpful if a change like this was mentioned in the release notes.

Reply:
> ```
> func init() {
> 	err := os.MkdirAll(path.Join(utils.GetHome(), ".meshery", "bin"), 0750)
> 	if err != nil {
> 		fmt.Println(err)
> 		os.Exit(0)
> 	}
> }
> ```

Should this (i.e. a default `init` function) be a implemented in meshkit? 
What about the `.kube` directory?
These prerequisites should also be documented IMHO. E.g. creating and picking up `kubeconfig` in different places (sometimes hardcoded paths) is quite confusing and can break things.

Reply:
To be honest, I do not like these codes at all. I believe it was no reason we should init to create the folder(If do, we should list the reason why and try to optimization, this is the reason I suggested you do not upgrade to v0.1.7 ).  So, we take more time discuss this. But for current status, it's ok we keep this in this PR.